authentication/src/main/java/io/camunda/authentication/csrf/CsrfProtectionMatcherWithExemptedUrls.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.csrf;
+
+import jakarta.servlet.http.HttpServletRequest;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+public class CsrfProtectionMatcherWithExemptedUrls implements RequestMatcher {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(CsrfProtectionMatcherWithExemptedUrls.class);
+
+  // CSRF-safe methods
+  private static final Pattern ALLOWED_METHODS = Pattern.compile("^(GET|HEAD|TRACE|OPTIONS)$");
+
+  private final Pattern allowedPathsPattern;
+
+  public CsrfProtectionMatcherWithExemptedUrls(final Set<String>... allowedPathsSets) {
+    final Set<String> allowedPaths = new HashSet<>();
+    for (final Set<String> allowedPathSet : allowedPathsSets) {
+      allowedPaths.addAll(allowedPathSet);
+    }
+    allowedPathsPattern = allowedPathsToPattern(allowedPaths);
```

Comment:
❓ Why must the allowed paths be passed? In the old implementation, they had been hard-coded. Would this be enough?